hack/release/common.sh

```diff
@@ -0,0 +1,157 @@
+  img_tag="$(echo "${img}" | cut -d "@" -f 1 | cut -d ":" -f 2 -s)"
+  img_digest="$(echo "${img}" | cut -d "@" -f 2)"
+
+  img_registry="$(echo "${img_repo}" | cut -d "/" -f 1)"
+  img_path="$(echo "${img_repo}" | cut -d "/" -f 2-)"
+
+  lockImage "${img_registry}" "${img_path}" "${img_tag}"
+  lockImage "${img_registry}" "${img_path}" "${img_tag}-aks"
+
+  cosignOciArtifact "${version}" "${commit_sha}" "${build_date}" "${img}"
+  cosignOciArtifact "${version}" "${commit_sha}" "${build_date}" "${img_nap}"
+
+  repo=${5:-$img_repo} # override the repo if provided (used for MCR)
+
+  yq e -i ".controller.image.repository = \"${repo}\"" charts/karpenter/values.yaml
+  yq e -i ".controller.image.tag = \"${img_tag}\"" charts/karpenter/values.yaml
+  yq e -i ".controller.image.digest = \"${img_digest}\"" charts/karpenter/values.yaml
+
+  publishHelmChart "${oci_repo}" "karpenter" "${helm_chart_version}" "${commit_sha}" "${build_date}"
+  publishHelmChart "${oci_repo}" "karpenter-crd" "${helm_chart_version}" "${commit_sha}" "${build_date}"
+}
+
+lockImage() {
+  local img_registry img_path img_tag
+
+  img_registry="$1"
+  img_path="$2"
+  img_tag="$3"
+
+	az acr repository update -n "${img_registry}" --image "${img_path}:${img_tag}" \
+    --write-enabled false \
+    --delete-enabled false
+}
+
+publishHelmChart() {
+  local oci_repo helm_chart version commit_sha build_date helm_chart_artifact helm_chart_digest
+
+  oci_repo="${1}"
+  helm_chart="${2}"
+  version="${3}"
+  commit_sha="${4}"
+  build_date="${5}"
+
+  helm_chart_artifact="${helm_chart}-${version}.tgz"
+
+  yq e -i ".appVersion = \"${version}\"" "charts/${helm_chart}/Chart.yaml"
+  yq e -i ".version = \"${version}\"" "charts/${helm_chart}/Chart.yaml"
+
+  cd charts
+  helm dependency update "${helm_chart}"
+  helm lint "${helm_chart}"
+  helm package "${helm_chart}" --version "${version}"
+  helm push "${helm_chart_artifact}" "oci://${oci_repo}"
+  rm "${helm_chart_artifact}"
+  cd ..
+
+  helm_chart_digest="$(crane digest "${oci_repo}/${helm_chart}:${version}")"
+  cosignOciArtifact "${version}" "${commit_sha}" "${build_date}" "${oci_repo}${helm_chart}:${version}@${helm_chart_digest}"
+}
+
+# When executed interactively, cosign will prompt you to authenticate via OIDC, where you'll sign in
+# with your email address. Under the hood, cosign will request a code signing certificate from the Fulcio
+# certificate authority. The subject of the certificate will match the email address you logged in with.
+# Cosign will then store the signature and certificate in the Rekor transparency log, and upload the signature
+# to the OCI registry alongside the image you're signing. For details see https://github.com/sigstore/cosign.
+cosignOciArtifact() {
+  local version commit_sha build_date artifact
+
+  version="${1}"
+  commit_sha="${2}"
+  build_date="${3}"
+  artifact="${4}"
+
+  cosign sign --yes -a version="${version}" -a commitSha="${commit_sha}" -a buildDate="${build_date}" "${artifact}"
```

Comment:
cosign is probably worthy of mentioning in the README.md Install utilities section